commons/ava_testsuite/staking_gecko_network_tests.go

```diff
@@ -125,159 +120,70 @@ func (test StakingNetworkRpcWorkflowTest) GetTimeout() time.Duration {
 
 
 // =================== Fully Connected Test ==============================
-type FiveNodeStakingNetworkFullyConnectedTest struct{}
-func (test FiveNodeStakingNetworkFullyConnectedTest) Run(network interface{}, context testsuite.TestContext) {
+type StakingNetworkFullyConnectedTest struct{}
+func (test StakingNetworkFullyConnectedTest) Run(network interface{}, context testsuite.TestContext) {
 	castedNetwork := network.(ava_networks.TestGeckoNetwork)
+	normalServiceId1 := 0
+	normalServiceId2 := 1
 
-	allServiceIds := castedNetwork.GetAllBootServiceIds()
-	allServiceIds[NODE_SERVICE_ID] = true
+	stakerIds := castedNetwork.GetAllBootServiceIds()
+	allServiceIds := make(map[int]bool)
+	for stakerId, _ := range stakerIds {
+		allServiceIds[stakerId] = true
+	}
+	// Add our custom nodes
+	allServiceIds[normalServiceId1] = true
+	allServiceIds[normalServiceId2] = true
 
-	extraStakerClient, err := castedNetwork.GetGeckoClient(NODE_SERVICE_ID)
-	if err != nil {
-		context.Fatal(stacktrace.Propagate(err, "Failed to get extra staker client."))
+	allNodeIds, allGeckoClients := getNodeIdsAndClients(context, castedNetwork, allServiceIds)
+	if err := verifyNetworkFullyConnected(context, allServiceIds, stakerIds, allNodeIds, allGeckoClients); err != nil {
+		context.Fatal(stacktrace.Propagate(err, "An error occurred verifying the network's state"))
 	}
+
+	normalServiceClient1 := allGeckoClients[normalServiceId1]
 	highLevelExtraStakerClient := ava_networks.NewHighLevelGeckoClient(
-		extraStakerClient,
+		normalServiceClient1,
 		STAKER_USERNAME,
 		STAKER_PASSWORD)
-	err = highLevelExtraStakerClient.GetFundsAndStartValidating(SEED_AMOUNT, STAKE_AMOUNT)
-	if err != nil {
+	if err := highLevelExtraStakerClient.GetFundsAndStartValidating(SEED_AMOUNT, STAKE_AMOUNT); err != nil {
```

Comment:
@mieubrisse I believe we'll have to add _both_ non-bootstrapper nodes as  validators in order for this to work

Reply:
actually, could be wrong. not sure whether a non-staker node will participate in gossiping to grab more peers. but for sure a validator node participates in the gossip and thus picks up peers

Reply:
Yep, we don't need to